src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java

```diff
@@ -120,45 +126,55 @@ public AnswerController(AnswerService answerService, QuestionService questionSer
 
-        AnswerResponse answerResponse = new AnswerResponse();
-        try {
-            Question question = new Question();
-            question.setQuestionId(questionId);
-            Answer answer = new Answer().answerId(answerId)
-                    .answerText(request.getAnswer().getAnswer());
-            answer.setQuestion(question);
+        ValidationResult validationResult = validate(request);
+        if (!validationResult.isValid()) {
+            return new ResponseEntity<AnswerResponse>(HttpStatus.UNPROCESSABLE_ENTITY);
+        }
 
-            Optional<Answer> optionalAnswer = answerService.retrieveAnswerById(answerId);
-            if (!optionalAnswer.isPresent()) {
-                return new ResponseEntity<AnswerResponse>(HttpStatus.NOT_FOUND);
-            }
+        Optional<Answer> optAnswer = answerService.retrieveAnswerById(answerId);
 
-            answerService.updateAnswerById(answer);
-            answerResponse.setAnswerId(answer.getAnswerId());
-        } catch (Exception e) {
+        if(!optAnswer.isPresent()){
+            return new ResponseEntity<AnswerResponse>(HttpStatus.NOT_FOUND);
+        }
+
+        Optional<AnswerState> optionalAnswerState = answerStateService.retrieveAnswerStateByState(request.getAnswerState());
+        if(!optionalAnswerState.isPresent()){
             return new ResponseEntity<AnswerResponse>(HttpStatus.FAILED_DEPENDENCY);
         }
 
+        Answer body = new Answer();
+        body.setAnswerState(optionalAnswerState.get());
+        body.setAnswerText(request.getAnswerText());
+        Answer updatedAnswer = answerService.updateAnswer(optAnswer.get(), body);
+
+        AnswerResponse answerResponse = new AnswerResponse();
+        answerResponse.setAnswerId(updatedAnswer.getAnswerId());
+
         return ResponseEntity.ok(answerResponse);
     }
 
     private ValidationResult validate(AnswerRequest request) {
         ValidationResult result = new ValidationResult();
         result.setValid(true);
 
-        if (request.getAnswer() == null || StringUtils.isEmpty(request.getAnswer().getAnswer())) {
+        if (request.getAnswerText() == null || StringUtils.isEmpty(request.getAnswerText())) {
             result.setValid(false);
             result.setReason("Answer text cannot be empty");
-        } else if (request.getAnswerState() == null || StringUtils.isEmpty(request.getAnswerState().getStateName())) {
+        } else if (StringUtils.isEmpty(request.getAnswerState())) {
             result.setValid(false);
             result.setReason("Answer state cannot be empty");
+        } else {
+            Optional<AnswerState> optAnswerState = answerStateService.retrieveAnswerStateByState(request.getAnswerState());
+            if (!optAnswerState.isPresent()) {
+                result.setValid(false);
+                result.setReason("Answer state is not valid");
+            }
```

Comment:
In addition, it should not be possible to transition from SUBMITTED to DRAFT.

Reply:
I can add code to do this check but I thought we were going to use a state transition table.

Reply:
At the mo, the application isn't complicated enough to justify a state transition table just to support 2 states. If it gets more complicated than this we should certainly as the designer for it.